R/weight.R

```diff
@@ -100,57 +99,54 @@ makeWeight <- function(..., name) {
     out
 }
 
-#' Validate an expression passed to makeWeight
+#' Generate entry for [makeWeight]
 #'
-#' Utility function to catch formula errors passed to [makeWeight].
+#' Utility function to catch formula errors passed to [makeWeight], and to generate
+#' the appropriate entry.
 #'
 #' @param expr
 #' An expression
 #' @return
-#' NULL value, function is called for its side effects
+#' A list of the variable id and the target weights.
 #'
 #' @keywords internal
 #' @examples
 #' \dontrun{
 #' validateWeightExpression(ds$var ~ c(10, 30, 60))
 #' }
 #'
-validateWeightExpression <- function(expr) {
-    if (length(expr) != 3) {
-        halt(paste(expr, "is an invalid expression, use the form ds$var ~ c(10, 20, 30)"))
+generateWeightEntry <- function(expr) {
+    formula <- try(as.formula(expr), silent = TRUE)
+    if (is.error(formula)) {
+        halt(expr, " is not a valid formula, use the form ds$var ~ c(10, 20, 30)")
```

Comment:
I wonder if you should do `substitute(expr)` here (possibly with an `as.character()` around it, just in case (and also possibly adding in a test using a non-character object to the tests below. This way the user sees what they typed.